### PR TITLE
Add support for AWSCLI_LOGIN_ROOT

### PR DESCRIPTION
**Pull request by ddriddle**

### CODE REVIEW

File: src/awscli_login/config.py [hunk 1]
```diff
@@ -105,17 +105,17 @@ class Profile:
 
     def _init_dir(self) -> None:
         """ Create ~/.aws-login directory if it does not exist. """
-        home = expanduser('~')
+        root = environ.get('AWSCLI_LOGIN_ROOT')
 
-        self.home = home
+        self.home = root if root is not None else expanduser('~')
```
Comment (JonRoma) [inline src/awscli_login/config.py:110]: If `AWSCLI_LOGIN_ROOT` is an empty string, `self.home` will be set to that value, causing an empty string to later be concatenated with various directories in the code at lines 113-118. Is this really what you want?

File: src/tests/base.py [hunk 2]
```diff
@@ -398,7 +398,37 @@ def assertTmpFileNotTouchedBy(self, filename: str, func: Callable,
                                  relpath(filename, self.tmpd.name))
 
 
-class CleanAWSLoginEnvironment(TempDir):
+class CleanEnvironment(unittest.TestCase):
+    """Mixin Class for clearing and setting environment variables."""
```
Comment (JonRoma) [inline src/tests/base.py:402]: Interesting code. I don't take as much advantage of mixin classes as I would like.

Comment (JonRoma) [inline src/tests/base.py:402]: Interesting code. I don't take as much advantage of mixin classes as I would like.

Comment (JonRoma) [inline src/tests/base.py:402]: Interesting code. I don't take as much advantage of mixin classes as I would like.

File: src/awscli_login/config.py (hunk 1 above)
Comment (edthedev) [reply src/awscli_login/config.py:110]:
```suggestion
self.home = root if root else expanduser('~')
```

Good catch Jon.

**Review by edthedev — APPROVED** [review]

File: src/awscli_login/config.py (hunk 1 above)
Comment (ddriddle) [reply src/awscli_login/config.py:110]: Yes, I do. It allows users to use relative paths. It's an edge case I would not encourage but somebody may find it useful. Plus I'm not really doing any error checking to verify that AWSCLI_LOGIN_ROOT is an actual valid path to a directory. That would be the solution. So in short, an empty string is as valid as anything else from the point of view of this portion of the code.

File: src/tests/base.py (hunk 2 above)
Comment (ddriddle) [reply src/tests/base.py:402]: Mixins come in handy. The test code is very classy and complex. I'm happy that I made the most of the class system from a learning perspective, but from a maintenance perspective I wish I had used [pytest](https://docs.pytest.org/en/7.1.x/) which is much more pythonic then the built in unittest framework which feels like java. [pytest](https://docs.pytest.org/en/7.1.x/) is based on functions and is just much easier to deal with even with dealing with complex tests where you want to keep everything nice and DRY.

**Review by tzturner — APPROVED** [review]

**Pull request merged**